### DISCUSSION TITLE
Can I use Speaker Recognition with Speaker Embeddings?

### DISCUSSION THREAD

**Discussion by louismorgner** [speechbrain/speechbrain#1769] [Q&A] [answered]
I came across the [Speaker Verification with ECAPA-TDNN embeddings](https://huggingface.co/speechbrain/spkrec-ecapa-voxceleb) and was curious whether there is any way to achieve the following:

1. Compute speaker embeddings with the EncoderClassifier
2. Verify the speaker against this embedding instead of another file

Any hints on how I could achieve this would be highly appreciated. Thanks!

**Comment by asumagic** [comment, marked as answer]
If I got you right (i.e. you want to check against an existing embedding rather than have to provide the original waveform), I believe it should be easy enough. See `SpeakerRecognition` source code in `interfaces.py`: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/pretrained/interfaces.py#L1022

In particular, all the actual verification does is:

```python
emb1 = self.encode_batch(wavs1, wav1_lens, normalize=True)
emb2 = self.encode_batch(wavs2, wav2_lens, normalize=True)
score = self.similarity(emb1, emb2)
return score, score > threshold
```

Where `self.similarity` is simply:

```python
self.similarity = torch.nn.CosineSimilarity(dim=-1, eps=1e-6)
```

So you could simply use `spk_rec.similarity(emb1, emb2)` over your pre-computed embeddings. `SpeakerRecognition` inherits from `EncoderClassifier`, so you can use `encode_batch` from that same class.

**Reply by louismorgner** [reply]
That's exactly what I was looking for. Thank you so much @AsuMagic. Will give it a shot right away.
